instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/instrumentation/scheduling/InternalThreadAttributesExtractor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.internal.instrumentation.scheduling;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.semconv.incubating.ThreadIncubatingAttributes;
+import javax.annotation.Nullable;
+import org.springframework.lang.NonNull;
+
+/** Expose current thread to attributes. */
+class InternalThreadAttributesExtractor<REQUEST, RESPONSE>
+    implements AttributesExtractor<REQUEST, RESPONSE> {
+  @Override
+  public void onStart(
+      @NonNull AttributesBuilder attributes,
+      @NonNull Context parentContext,
+      @NonNull REQUEST classAndMethod) {
+    attributes
+        .put(ThreadIncubatingAttributes.THREAD_ID, Thread.currentThread().getId())
```

Comment:
Is the same thing done with the Java agent instrumentation?

Reply:
The final result is quite the same as java agent, but it's implemented differently. Java agent uses `AddThreadDetailsSpanProcessor` to add `thread.id` to the attributes of a span, while in spring boot auto configuration this information seems missing in span. Hence, to match the final span as java agent I write this.
For further extension, this class may be extracted as a common span processor and configured automatically just like the one in java agent?

Reply:
Thank you for your explanations. The current implementation is good to me.

Reply:
> For further extension, this class may be extracted as a common span processor and configured automatically just like the one in java agent?

I think this is a good idea 👍

I'd recommend removing it from this PR (since in the current state it's currently inconsistent with the other spring starter instrumentation), then we can merge, and then you can send a follow-up PR for this

Reply:
That sounds cool to me😊. I've made the change.